WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductCoordinator.swift

```diff
@@ -98,8 +103,7 @@ final class AddProductCoordinator: Coordinator {
         ServiceLocator.analytics.track(event: .ProductCreation.addProductStarted(source: source,
                                                                                  storeHasProducts: storeHasProducts))
 
-        // TODO-10688: Replace this with eligibility check
-        let isEligibleForAI = ServiceLocator.featureFlagService.isFeatureFlagEnabled(.productCreationAI)
+        let isEligibleForAI = addProductWithAIEligibilityChecker.isEligible(storeHasProducts: storeHasProducts)
```

Comment:
👀 We should also update `storeHasProduct` to filter out sample products for free trial stores [pecCkj-Zo-p2#comment-506]

Reply:
This ☝️ requires updating Networking and Yosemite, I'll take care of that.

Reply:
Thank you! 🙇